pkg/gateway/model_build_targets.go

```diff
@@ -111,19 +112,38 @@ func (t *latticeTargetsModelBuildTask) buildLatticeTargets(ctx context.Context)
 		return errors.New(errmsg)
 	}
 
-	portAnnotations := undefinedPort
-	serviceExport := &mcs_api.ServiceExport{}
-	err = t.Client.Get(ctx, namespacedName, serviceExport)
-	if err != nil {
-		glog.V(6).Infof("Failed to find Service export in the DS. Name:%v, Namespace:%v - err:%s\n ", t.tgName, t.tgNamespace, err)
-	} else {
-		// TODO: Change the code to support multiple comma separated ports instead of a single port
-		//portsAnnotations := strings.Split(serviceExport.ObjectMeta.Annotations["multicluster.x-k8s.io/Ports"], ",")
-		portAnnotations, err = strconv.ParseInt(serviceExport.ObjectMeta.Annotations[portAnnotationsKey], 10, 64)
+	definedPorts := make([]int64, undefinedPort)
+	if tg.ByServiceExport {
+		serviceExport := &mcs_api.ServiceExport{}
+		err = t.Client.Get(ctx, namespacedName, serviceExport)
 		if err != nil {
-			glog.V(6).Infof("Failed to read Annotaions/Port:%v, err:%s\n ", serviceExport.ObjectMeta.Annotations[portAnnotationsKey], err)
+			glog.V(6).Infof("Failed to find Service export in the DS. Name:%v, Namespace:%v - err:%s\n ", t.tgName, t.tgNamespace, err)
+		} else {
+			portsAnnotations := strings.Split(serviceExport.ObjectMeta.Annotations[portAnnotationsKey], ",")
+
+			for _, portAnnotation := range portsAnnotations {
+				definedPort, err := strconv.ParseInt(portAnnotation, 10, 64)
+				if err != nil {
+					glog.V(6).Infof("Failed to read Annotations/Port:%v, err:%s\n ", serviceExport.ObjectMeta.Annotations[portAnnotationsKey], err)
+				}
+				definedPorts = append(definedPorts, definedPort)
+			}
+			glog.V(6).Infof("Build Targets - portAnnotations: %v \n", definedPorts)
+		}
```

Comment:
You could encapsulate line 122-132 into a helper function to make code more clear:
```
func getServiceExportAnnotationDefinedPorts(serviceExport *mcs_api.ServiceExport) []int64 {
	if serviceExport.ObjectMeta.Annotations == nil {
		return []int64{}
	}
	portsAnnotations := strings.Split(serviceExport.ObjectMeta.Annotations[portAnnotationsKey], ",")
	var definedPorts []int64
	for _, portAnnotation := range portsAnnotations {
		definedPort, err := strconv.ParseInt(portAnnotation, 10, 64)
		if err != nil {
			glog.V(6).Infof("Failed to read Annotations/Port:%v, err:%s\n ", serviceExport.ObjectMeta.Annotations[portAnnotationsKey], err)
		}
		definedPorts = append(definedPorts, definedPort)
	}
	return definedPorts
}
```